### PR TITLE
Add Variant and genotype call bdg to ga4gh converters

### PR DESCRIPTION
**Pull request by jpdna**
I'm going to check this over a bit more for formatting and add tests with more combinations of genotypes, but wanted to share now.  

Works so far, tests pass.

If you have any comments now they are welcome!

### CODE REVIEW

**Comment by akmorrow13** [comment]
hi @jpdna ! This looks good. Are we still waiting on more tests for this?

**Comment by heuermh** [comment]
@jpdna ping when ready to review

**Comment by jpdna** [comment]
If you don't mind, please go ahead and review it @heuermh , and if you have comments on where you think more tests would are needed or desirable please point out. Thanks!

**Comment by heuermh** [comment]
> ...if you have comments on where you think more tests would are needed or desirable please point out.

We don't have CI on this repo yet, meanwhile you can run
```
$ mvn cobertura:cobertura
```
to generate unit test coverage reports, found in e.g. `convert-ga4gh/target/site/cobertura/index.html`.

File: convert-ga4gh/src/main/java/org/bdgenomics/convert/ga4gh/BdgenomicsGenotypeToGa4ghCall.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
+package org.bdgenomics.convert.ga4gh;
```
Comment (heuermh) [inline convert-ga4gh/src/main/java/org/bdgenomics/convert/ga4gh/BdgenomicsGenotypeToGa4ghCall.java]: needs license header

File: convert-ga4gh/src/main/java/org/bdgenomics/convert/ga4gh/BdgenomicsGenotypeToGa4ghCall.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    @Override
+    public ga4gh.Variants.Call convert(final org.bdgenomics.formats.avro.Genotype genotype,
+                                       final ConversionStringency stringency,
+                                       final Logger logger) throws ConversionException {
+        if (genotype == null) {
+            warnOrThrow(genotype, "must not be null", null, stringency, logger);
+            return null;
+        }
+
+        java.util.List<GenotypeAllele> inputAlleles = genotype.getAlleles();
```
Comment (heuermh) [inline convert-ga4gh/src/main/java/org/bdgenomics/convert/ga4gh/BdgenomicsGenotypeToGa4ghCall.java]: this code block should be a separate converter

File: convert-ga4gh/src/main/java/org/bdgenomics/convert/ga4gh/BdgenomicsGenotypeToGa4ghCall.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        String alleleFirst = toGA4GHAllele(inputAlleles.get(0));
+        String alleleSecond = "";
+        if (inputAlleles.size() == 2) {
+            alleleSecond = toGA4GHAllele(inputAlleles.get(1));
+        } else {
+            alleleSecond = ".";
+        }
+
+        com.google.protobuf.Value alleleFirstValue = com.google.protobuf.Value.newBuilder().setStringValue(alleleFirst).build();
+        com.google.protobuf.Value alleleSecondtValue = com.google.protobuf.Value.newBuilder().setStringValue(alleleSecond).build();
```
Comment (heuermh) [inline convert-ga4gh/src/main/java/org/bdgenomics/convert/ga4gh/BdgenomicsGenotypeToGa4ghCall.java]: alleleSecondtValue &rarr; alleleSecondValue

File: convert-ga4gh/src/main/java/org/bdgenomics/convert/ga4gh/BdgenomicsVariantToGa4ghVariant.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
+package org.bdgenomics.convert.ga4gh;
```
Comment (heuermh) [inline convert-ga4gh/src/main/java/org/bdgenomics/convert/ga4gh/BdgenomicsVariantToGa4ghVariant.java]: needs license header

File: convert-ga4gh/src/main/java/org/bdgenomics/convert/ga4gh/BdgenomicsVariantToGa4ghVariant.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+/**
```
Comment (heuermh) [inline convert-ga4gh/src/main/java/org/bdgenomics/convert/ga4gh/BdgenomicsVariantToGa4ghVariant.java]: this comment block should be removed

File: convert-ga4gh/src/test/java/org/bdgenomics/convert/ga4gh/BdgenomicsGenotypeToGa4ghCallTest.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
+package org.bdgenomics.convert.ga4gh;
```
Comment (heuermh) [inline convert-ga4gh/src/test/java/org/bdgenomics/convert/ga4gh/BdgenomicsGenotypeToGa4ghCallTest.java]: needs license header

File: convert-ga4gh/src/test/java/org/bdgenomics/convert/ga4gh/BdgenomicsVariantToGa4ghVariantTest.java [hunk 7]
```diff
@@ -0,0 +1,78 @@
+package org.bdgenomics.convert.ga4gh;
```
Comment (heuermh) [inline convert-ga4gh/src/test/java/org/bdgenomics/convert/ga4gh/BdgenomicsVariantToGa4ghVariantTest.java]: needs license header

File: convert-ga4gh/src/test/java/org/bdgenomics/convert/ga4gh/BdgenomicsVariantToGa4ghVariantTest.java [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+    public void testConvertNullSilent() {
+        assertNull(variantConverter.convert(null, ConversionStringency.SILENT, logger));
+    }
+
+    @Test
+    public void testConvert() {
+        ga4gh.Variants.Variant ga4ghVariant = variantConverter.convert(bdgVariant, ConversionStringency.STRICT, logger);
+        assertEquals("1", ga4ghVariant.getReferenceName());
+        assertEquals( 19190L, ga4ghVariant.getStart());
+        assertEquals(19190L, ga4ghVariant.getEnd());
```
Comment (heuermh) [inline convert-ga4gh/src/test/java/org/bdgenomics/convert/ga4gh/BdgenomicsVariantToGa4ghVariantTest.java]: start and end shouldn't be equal, no?  :)

File: convert-ga4gh/src/main/java/org/bdgenomics/convert/ga4gh/BdgenomicsVariantToGa4ghVariant.java [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+                .setReferenceBases(variant.getReferenceAllele())
+                .addAllAlternateBases(Arrays.asList(variant.getAlternateAllele() ))
+                .addAllFiltersFailed(variant.getFiltersFailed())
+                .setFiltersPassed(variant.getFiltersPassed())
+                .setFiltersApplied(variant.getFiltersApplied())
+                .build();
+    }
+}
+
+
```
Comment (akmorrow13) [inline convert-ga4gh/src/main/java/org/bdgenomics/convert/ga4gh/BdgenomicsVariantToGa4ghVariant.java]: -newline

**Comment by jpdna** [comment]
@heuermh - please see if you are happy with the new `GenotypeAlleleToString` converter in addressing your earlier comment, or let me know if you meant something else.  Thanks!

File: convert-ga4gh/src/main/java/org/bdgenomics/convert/ga4gh/GenotypeAlleleToString.java [hunk 10]
```diff
@@ -0,0 +1,30 @@
+package org.bdgenomics.convert.ga4gh;
```
Comment (heuermh) [inline convert-ga4gh/src/main/java/org/bdgenomics/convert/ga4gh/GenotypeAlleleToString.java]: needs license header

**Comment by heuermh** [comment]
I was thinking `List<GenotypeAllele>` to `ListValue` maybe, but it doesn't matter too much.  One last missing header and then I'll squash and merge this in.

**Comment by jpdna** [comment]
> One last missing header and then I'll squash and merge this in.

done and squashed.

**Comment by akmorrow13** [comment]
+1

**Comment by heuermh** [comment]
Thank you, @jpdna!

**Pull request merged**